src/java_bytecode/java_types.h

```diff
@@ -374,4 +374,28 @@ void get_dependencies_from_generic_parameters(
   const typet &,
   std::set<irep_idt> &);
 
+class java_specialised_generic_class_typet : public java_generic_class_typet
+{
+public:
+  java_specialised_generic_class_typet()
+  {
+    set(ID_C_specialised_generic_java_class, true);
+  }
+
+  ~java_specialised_generic_class_typet() {}
+};
+
+inline const bool java_is_specialised_generic_class_type(const typet &type)
+{
+  return type.get_bool(ID_C_specialised_generic_java_class);
+}
+
+inline java_specialised_generic_class_typet
```

Comment:
Write const and non-const versions, and return by reference like all the rest of these cast functions